tensorflow_probability/python/distributions/two_piece_normal_test.py

```diff
@@ -348,6 +350,55 @@ def sampler(loc, scale, skewness):
       self.assertIsNotNone(grads[1])  # d/d scale
       self.assertIsNotNone(grads[2])  # d/d skewness
 
+  @test_util.numpy_disable_gradient_test
+  def testDifferentiableSampleNumerically(self):
+    """Test the gradients of the samples w.r.t. skewness."""
+    sample_shape = [int(2e5)]
+    seed = test_util.test_seed()
+
+    def sampler(skewness):
+      loc = tf.constant(0., self.dtype)
+      scale = tf.constant(1., self.dtype)
+      dist = tfd.TwoPieceNormal(
+          loc, scale=scale, skewness=skewness, validate_args=True)
+      return tf.reduce_mean(dist.sample(sample_shape, seed=seed))
+
+    for skewness in [0.75, 1., 1.33]:
+      err = self.compute_max_gradient_error(
+          sampler, [tf.constant(skewness, self.dtype)], delta=0.1)
+      self.assertLess(err, 0.05)
+
+  @test_util.numpy_disable_gradient_test
+  def testDifferentiableSampleAnalytically(self):
+    """Test the gradients of the samples w.r.t. loc and scale."""
+    n = 100
+    sample_shape = [n, n]
+    n_samples = np.prod(sample_shape)
+
+    n_params = 20
+    loc = tf.constant(
+        np.linspace(-3., stop=3., num=n_params), dtype=self.dtype)
+    scale = tf.constant(
+        np.linspace(0.1, stop=10., num=n_params), dtype=self.dtype)
+    skewness = tf.constant(
+        np.linspace(0.75, stop=1.33, num=n_params), dtype=self.dtype)
+
+    seed = test_util.test_seed()
+
+    def sampler(loc, scale):
+      dist = tfd.TwoPieceNormal(
+          loc=loc, scale=scale, skewness=skewness, validate_args=True)
+      return dist.sample(sample_shape, seed=seed)
```

Comment:
Same non-linearity suggestion here as well (although here you want something that's easy to analytically handle).

Reply:
Here I used the nonlinear function `tf.math.exp`.